### PR TITLE
Create guess_the_number_search.py

### PR DESCRIPTION
**Pull request by hakiKhuva**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by pronoym99** [comment]
@hakiKhuva ```Run tests``` is currently failing due to [another algorithm](https://github.com/TheAlgorithms/Python/blob/master/machine_learning/xgboost_classifier.py). The actions in this repository end up building the whole directory and running the tests for all the algorithms for **every commit**. I don't know why it is done like that. A similar thing is happening to my commit in #7811.

File: other/guess_the_number_search.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+    >>> temp_input_value(option=True)
+    10
+
+    >>> temp_input_value(option=False)
+    1000
+    """
+    if option is True:
+        return min_val
+    return max_val
```
Comment (cclauss) [inline other/guess_the_number_search.py:73]:
```suggestion
    return min_val if option else max_val
```

File: other/guess_the_number_search.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+
+def get_avg(number_1: int, number_2: int) -> int:
+    """
+    Return the mid-number(whole) of two integers a and b
+    >>> get_avg(10,15)
+    12
+    >>> get_avg(20,300)
+    160
+    >>> get_avg("abcd",300)
```
Comment (cclauss) [inline other/guess_the_number_search.py:83]:
```suggestion
    >>> get_avg(10, 15)
    12
    >>> get_avg(20, 300)
    160
    >>> get_avg("abcd", 300)
```

File: other/guess_the_number_search.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+def temp_input_value(min_val: int = 10, max_val: int = 1000, option: int = True) -> int:
+    """
+    Temporary input values for tests
+
+    >>> temp_input_value(option=True)
+    10
+
+    >>> temp_input_value(option=False)
+    1000
```
Comment (cclauss) [inline other/guess_the_number_search.py:69]: Please add tests for zero, negative numbers, positive numbers, max_val which is less then min_val.

File: other/guess_the_number_search.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+>>> guess_the_number(10, 1000, 17)
+started...
+guess the number : 17
+details : [505, 257, 133, 71, 40, 25, 17]
+
+"""
+
+
+def temp_input_value(min_val: int = 10, max_val: int = 1000, option: int = True) -> int:
```
Comment (cclauss) [inline other/guess_the_number_search.py:61]:
```suggestion
def temp_input_value(min_val: int = 10, max_val: int = 1000, option: bool = True) -> int:
```

File: other/guess_the_number_search.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+    >>> guess_the_number(10, 1000, 17)
+    started...
+    guess the number : 17
+    details : [505, 257, 133, 71, 40, 25, 17]
+
+    >>> guess_the_number(10, 1000, "a")
+    Traceback (most recent call last):
+        ...
+    TypeError: '>' not supported between instances of 'int' and 'str'
```
Comment (cclauss) [inline other/guess_the_number_search.py:104]: Please add tests for zero, negative numbers, and floating point numbers.

A ValueError must be raised if to_guess is less than lower or greater than upper or lower is greater than upper.

File: other/guess_the_number_search.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    AssertionError: Invalid type of value(s) specified to function!
+
+    >>> temp_input_value(min_val=-100, max_val=500)
+    -100
+
+    >>> temp_input_value(min_val=-5100, max_val=-100)
+    -5100
+    """
+    assert (
+        type(min_val) == int and type(max_val) == int and type(option) == bool
```
Comment (cclauss) [inline other/guess_the_number_search.py:93]: PEP8: Use `isinstance()` instead of directly comparing types.

File: other/guess_the_number_search.py [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    ValueError: argument value for lower and higher must be(lower > higher)
+    """
+    assert (
+        type(lower) == int and type(higher) == int and type(to_guess) == int
+    ), 'argument values must be type of "int"'
+
+    if lower > higher:
+        raise ValueError("argument value for lower and higher must be(lower > higher)")
+
+    if to_guess < lower or to_guess > higher:
```
Comment (cclauss) [inline other/guess_the_number_search.py:158]:
```suggestion
    if not lower < to_guess < higher:
```

File: other/guess_the_number_search.py [hunk 8]
```diff
@@ -89,11 +89,13 @@ def temp_input_value(
     >>> temp_input_value(min_val=-5100, max_val=-100)
     -5100
     """
-    assert (
-        type(min_val) == int and type(max_val) == int and type(option) == bool
-    ), "Invalid type of value(s) specified to function!"
+    assert isinstance(min_val, int) and isinstance(max_val, int) and \
+        isinstance(option, bool), \
```
Comment (cclauss) [inline other/guess_the_number_search.py]: PEP8: No backslashes in Python code.  Line continuation should be done with ( and ) like it was.

Comment (hakiKhuva) [reply other/guess_the_number_search.py]: If I don't use backslash then the single line length is more than 80 characters, should I use `if` statement with or?

Comment (cclauss) [reply other/guess_the_number_search.py]: assert (
    a,
    b,
    c), "You gotta give me something to work with..."

Comment (hakiKhuva) [reply other/guess_the_number_search.py]: as you suggested, the code is showing the following (flake8)

Screenshot (234)

Can I use as showed in following code

```python
assert (
        isinstance(min_val, int) and
        isinstance(max_val, int) and
        isinstance(option, bool)
    ), "Invalid type of value(s) specified to function!"
```

Comment (cclauss) [reply other/guess_the_number_search.py]: Perfect.

**Review by cclauss — APPROVED** [review]
Thanks for doing this.

File: other/guess_the_number_search.py (hunk 8 above)
Comment (hakiKhuva) [reply other/guess_the_number_search.py]: build/ checks are still failing.

File: other/guess_the_number_search.py [hunk 9]
```diff
@@ -1,51 +1,51 @@
 """
-Get the random number guessed by the computer by passing the lower, higher, and the
```
Comment (cclauss) [inline other/guess_the_number_search.py]: 88 characters per line max and sentences please.

Comment (hakiKhuva) [reply other/guess_the_number_search.py]: Sir, please check the new pushed commit, I've changed the sentences and documentation and  set to the maximum 88 characters

**Comment by pronoym99** [comment]
I am also not sure about this. Even my PR #7811 has not been merged yet although the concerned checks have passed and I have done the necessary changes.

**Review by ChrisO345 — APPROVED** [review]
